enumerable.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+module Enumerable
+  def my_each()
+    return self unless block_given?
+
+    length.times do |index|
+      yield(self[index])
+    end
+  end
```

Comment:
- [ ] This should return an Enum if no block is given
![image](https://user-images.githubusercontent.com/61479648/90058134-b6aae380-dcb7-11ea-9e9c-6b6463019246.png)

- [ ] It should return self if block is given

- [ ] when a block is given and when self is a range, returns the range itself after calling the given block once for each element in self

- [ ] when a block is given and when self is a range, it behaves like #each and calls the given block once for each element in self

- [ ] when a block is given when self is a hash, it behaves like #each and calls the given block once for each element in self